src/components/Navbar.jsx

```diff
@@ -0,0 +1,139 @@
+  const translationMenuRef = useRef(null);
+  const hamburgerMenuRef = useRef(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (
+        !translationMenuRef.current?.contains(event.target) &&
+        !hamburgerMenuRef.current?.contains(event.target)
+      ) {
+        setIsTranslationMenuOpen(false);
+        setIsHamburgerMenuOpen(false);
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
+
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [translationMenuRef, hamburgerMenuRef]);
+
+  return (
+    <nav className={className}>
+      <div className="container flex justify-between items-center px-6 text-white rounded-full shadow-2xl bg-black/75 font-heading">
+        <div className="my-3 ml-12 sm:ml-16">
+          <img
+            alt="Arise logo"
+            className="absolute -translate-x-16 sm:-translate-x-20 -translate-y-2/5 size-16 sm:size-20"
+            src={logoSrc}
+          />
+          <p className="font-bold text-orange-500 uppercase sm:text-xl">
+            Arise
+          </p>
+        </div>
+        <div className="hidden gap-x-10 text-sm lg:flex">
+          {sections.map(({ title, clickHandler }) => (
+            <button key={title} onClick={clickHandler}>
+              <div className="relative ml-auto w-fit group">
+                <p>{title}</p>
+                <span className="hidden absolute w-full h-0.5 bg-gradient-to-r from-yellow-500 to-red-500 group-hover:block" />
+              </div>
+            </button>
+          ))}
+        </div>
+        <div className="flex gap-x-6">
+          {lngs && (
+            <div className="flex relative" ref={translationMenuRef}>
+              <button
+                onClick={() => {
+                  setIsTranslationMenuOpen(!isTranslationMenuOpen);
+                  setIsHamburgerMenuOpen(false);
+                }}
+                ref={translationMenuRef}
+              >
+                <BsTranslate size={24} />
+              </button>
+              {isTranslationMenuOpen && (
+                <div className="absolute -right-6 top-14 p-2 rounded-xl shadow-2xl bg-black/75">
+                  <ul className="w-fit">
+                    {lngs.map(({ FlagIcon, title, clickHandler }) => (
+                      <li
+                        className="flex gap-x-2 items-center py-1 px-2 text-sm rounded-lg hover:bg-gray-300/25"
+                        key={title}
+                        onClick={clickHandler}
+                      >
+                        <FlagIcon className="h-4" />
+                        <p>{title}</p>
+                      </li>
+                    ))}
+                  </ul>
+                </div>
+              )}
+            </div>
+          )}
+          <div className="flex relative lg:hidden" ref={hamburgerMenuRef}>
+            <button
```

Comment:
the button here is missing `hamburgerMenuRef`

Reply:
oh the `translationMenuRef` on the button above shouldn't be there, will remove. `ref`s should only reference one element: https://react.dev/reference/react/useRef

Reply:
okok, yeah that makes more sense

Reply:
i'll approve the PR, once you fix this

Reply:
there were other things that needed to be fixed, specifically the translation button. make sure to check the component in Storybook before approving